core/shared/src/main/scala/fs2/internal/InterruptContext.scala

```diff
@@ -88,15 +88,19 @@ final private[fs2] case class InterruptContext[F[_]](
     ref.get.flatMap {
       case Some(outcome) => F.pure(Left(outcome))
       case None =>
-        F.race(deferred.get, fa.attempt).map {
-          case Right(result) => result.leftMap(Outcome.Errored(_))
-          case Left(other)   => Left(other)
+        F.raceOutcome(deferred.get, fa.attempt).flatMap {
+          case Right(oc) =>
+            oc.embed(F.canceled.as(canceledError)).map(_.leftMap(Outcome.Errored(_)))
```

Comment:
Just a nit, maybe we can write this with `oc.fold(...)` to save the two separate ops?

Reply:
Oh yep, good call 👍 Will fix tomorrow